src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FallThroughCheck.java

```diff
@@ -266,7 +268,7 @@ public void visitToken(DetailAST ast) {
         if (!isLastGroup || checkLastCaseGroup) {
             final DetailAST slist = ast.findFirstToken(TokenTypes.SLIST);
 
-            if (slist != null && !isTerminated(slist, true, true)
+            if (slist != null && !isTerminated(slist, true, true, new HashSet<>())
```

Comment:
I don't use labels in Java code, but I noticed this is the only time we are "erasing" this labels list. We never seem to be removing anything from it, so things just keep growing.

Can we have 2 very distinct sets of code in the same case group use the same label names? For example, this compiles for me:
`````
        int a = 0;
        switch (a) {
        case 0:
            mylabel: while (a == 0) {
                break mylabel;
            }

            int b = 0;
            b++;

            mylabel: while (b == 0) {
                break mylabel;
            }
            break;
        default:
        }
`````
I also don't get any compile errors if a label is defined and never used.

Reply:
I dont get the point :)

This set is created every time we visit TokenTypes.CASE_GROUP, it is collected by GC after method invocation is completed. It doesn't really matter if labels are duplicated or not for this check. Same labels can be on same level, but not in hierarchy, e.g. this wont compile:
```
label_a: switch (1) {
        case 1: {
            label_a: break;
 }
```
we collect all labels inside case group and one set is enough for it

Reply:
Like I said, I don't really use labels so I am not sure if I can go further.

So it doesn't matter where the label is located in terms of the check? It just seems odd we are not taking these contexts into account and only capturing a partial look of the code.

Reply:
@strkkk , I still think we should not care  about labels at all.
read https://github.com/checkstyle/checkstyle/issues/11230#issuecomment-1055297662

I am complaetely ok to to make note in description that we do not care where label of break is pointing, if it is present we consider it as good code for FallThrough.

all madness with detection of infinite loop is out of this module consideration.

Reply:
I do not really understand what the problem is, to be honest.

> So it doesn't matter where the label is located in terms of the check? It just seems odd we are not taking these contexts into account and only capturing a partial look of the code. We have had bugs where we were not emptying caches correctly, so I am just checking if there is possible issues with your code.

This HashSet of labels is created for every CASE_GROUP. The idea is very simple - we only need to check label names inside case group because any other labelled break will mean that label is outside and code will exit somewhere outside CASE_GROUP. 

So, if we see something like `break my_cool_label` we only need to check if this `my_cool_label` in CASE_GROUP. If it is, there is no termination. If it is not - it is terminated.

@romani 
infinite loop is here just to make example shorter, since the only thing that matters is break.

Example code is weird, but it is compilable and is supported in this PR. I dont see much sense in reverting changes and correcting documentation

Reply:
I resolving this as I believe my concerns are not validated, but I still find it just odd that we don't care where the go to actually goes to and we are never doing any clean up.